environment.sh

```diff
@@ -36,52 +38,95 @@ done
     conda remove -n $env_name --all -y
-    env_exists="no"
+    create_env="yes"
   fi
+fi
 
-# Create new environment if it did not exist or we need to remake it
-if [ $env_exists == 'no' ]; then
-# This cannot be an else about because we set env_exists to no in that block
+if [ $create_env == 'yes' ]; then
   # Create new environment
   echo "Environment $env_name does not exist. Creating new environment $env_name..."
   # Create conda environment
   conda create -n $env_name python=3.11 -y
-  # Determine which requirements.txt to install from
-  if [ $env_type == 'simulation' ]; then
-    echo "Installing requirements for simulation environment..."
-    install_file="requirements.txt"
-  elif [ $env_type == 'artifact' ]; then
-    echo "Installing requirements for artifact environment"
-    install_file="artifact_requirements.txt"
-  else
-    echo "Invalid environment type. Valid argument types are simulation and artifact."
-    return 
-  fi
 fi
 
-# Activate new environment
+# Activate environment
 conda activate $env_name
+if [ $create_env == 'yes' ]; then
+  # Install json parser
+  conda install jq -y
+fi
+# At this point we have either a new environment or an environment that is less than a week old.
+# We need to check if we need to reinstall packages
+update_env=$create_env
+
+if [ $update_env == 'no' ]; then
+  # Check if environment was build before last modification to requirements file
+  if [[ $creation_time < $requirements_modification_time ]]; then
+    echo "Requirements file last modified after environment was created. Reinstalling packages..."
+    update_env="yes"
+  fi
+  # Check if there has been an update to vivarium packages since last modification to requirements file
+  grep @ $install_file 
+  # TODO: can we make this grep output a variable?
+  while read -r line ; do
+    # Parse each line of grep output
+    repo_info=(${line//@/ })
+    repo=${repo_info[0]}
+    repo_branch=${repo_info[2]}
+    last_commit_time=$(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/ihmeuw/$repo/commits?sha=$repo_branch | jq '.["0"].commit.committer.date')
+    if [[ $requirements_modification_time < $last_commit_time ]]; then
+      update_env="yes"
+      break
+    fi
+  # This way of writing/exiting while loop is a here string so the process runs
+  # in the main shell and not a subshell: https://www.gnu.org/software/bash/manual/bashref.html#Here-Strings
+  # I put an arbitrary empty string here but this is so we can set create_env to yes if we hit that trigger
+  done <<< "$(echo "")"
+fi
 
-# Install requirements via Github for new environment
-if [ $env_exists == 'no' ]; then
+# Install requirements via Github
+if [ $update_env == 'yes' ]; then
   # NOTE: update branch name if you update requirements.txt in a branch
-  pip install -r https://raw.githubusercontent.com/ihmeuw/vivarium_gates_nutrition_optimization_child/main/$install_file 
+  echo "Installing packages for $env_type environment"
```

Comment:
I don't understand the # NOTE above

Reply:
You can update the branch name to install from a requirements.txt from a specific branch on git hub